tests/conftest.py

```diff
@@ -235,7 +237,14 @@ def fanouthosts(ansible_adhoc, conn_graph_facts, creds):
         else:
             host_vars = ansible_adhoc().options['inventory_manager'].get_host(fanout_host).vars
             os_type = 'eos' if 'os' not in host_vars else host_vars['os']
-            fanout  = FanoutHost(ansible_adhoc, os_type, fanout_host, 'FanoutLeaf', creds['fanout_admin_user'], creds['fanout_admin_password'])
+            fanout  = FanoutHost(ansible_adhoc,
+                                 os_type,
+                                 fanout_host,
+                                 'FanoutLeaf',
+                                 creds['fanout_admin_user'],
+                                 creds["fanout_admin_password"],
+                                 shell_user=creds['fanout_root_user'] if 'fanout_root_user' in creds else None,
+                                 shell_passwd=creds['fanout_root_password'] if 'fanout_root_password' in creds else None)
```

Comment:
'fanout_admin_user' and 'fanout_admin_password' are credentials used for Arista bash shell. The network credentials are the ones used with 'switch_login' in ansible

Reply:
Since the credentials for accessing fanout is configurable, we don't have to stick with what we have in lab. I suggest to explicitly define these credentials in corresponding ansible group vars:
```
fanout_eos_user: <eos_username>
fanout_eos_password: <eos_password>
fanout_shell_user: <shell_username>
fanout_shell_password: <shell_password>
```
The `fanout_eos_user` and `fanout_eos_password` credential should expose the EOS CLI. The `fanout_shell_user` and `fanout_shell_password` credential should expose the Linux shell CLI.

But anyway, I have made some changes with the logic of getting fanout credentials. When the explicit credentials are not found, the code will fallback to use the `fanout_admin_user` and `fanout_admin_password`.